stac_fastapi/extensions/stac_fastapi/extensions/core/aggregation/types.py

```diff
@@ -2,33 +2,34 @@
 
 from typing import Any, Dict, List, Literal, Optional, Union
 
-from pydantic import Field
-from typing_extensions import TypedDict
+from typing_extensions import NotRequired, TypedDict
 
 from stac_fastapi.types.rfc3339 import DateTimeType
 
+Bucket = TypedDict(
+    "Bucket",
+    {
+        "key": str,
+        "data_type": str,
+        "frequency": NotRequired[Dict],
+        # we can't use the `class Bucket` notation because `from` is a reserved key
+        "from": NotRequired[Union[int, float]],
+        "to": NotRequired[Optional[Union[int, float]]],
+    },
+)
```

Comment:
I didn't caught that when reviewing the original PR but it was defining a type: TypedDict with Pydantic field 🤷